File: src/main/model/entity/LocalTag.ts
import BaseEntity from './BaseEntity.ts'

/**
 * 本地标签
 */
export default class LocalTag extends BaseEntity {
  /**
   * 主键
   */
  id: number | null | undefined
  /**
   * 本地标签名称
   */
  localTagName: string | null | undefined
  /**
   * 上级标签id
   */
  baseLocalTagId: number | null | undefined
  /**
   * 最后一次使用的时间
   */
  lastUse: number | null | undefined

  constructor(localTag?: LocalTag) {
    if (localTag === undefined) {
      super()
      this.id = undefined
      this.localTagName = undefined
      this.baseLocalTagId = undefined
      this.lastUse = undefined
    } else {
      super(localTag)
      this.id = localTag.id
      this.localTagName = localTag.localTagName
      this.baseLocalTagId = localTag.baseLocalTagId
      this.lastUse = localTag.lastUse
    }
  }
}

File: src/main-go/internal/model/local_tag.go
package model

// LocalTag 本地标签
type LocalTag struct {
	ID             int64  `gorm:"primaryKey;column:id" json:"id"`
	LocalTagName   string `gorm:"column:local_tag_name" json:"localTagName"`
	BaseLocalTagID int64  `gorm:"column:base_local_tag_id" json:"baseLocalTagId"`
	LastUse        int64  `gorm:"column:last_use" json:"lastUse"`
	CreateTime     int64  `gorm:"column:create_time" json:"createTime"`
	UpdateTime     int64  `gorm:"column:update_time" json:"updateTime"`
}

// TableName 指定表名
func (LocalTag) TableName() string {
	return "local_tag"
}

// GetID 实现 BaseEntity 接口
func (t LocalTag) GetID() int64 {
	return t.ID
}

// GetCreateTime 实现 BaseEntity 接口
func (t LocalTag) GetCreateTime() int64 {
	return t.CreateTime
}

// GetUpdateTime 实现 BaseEntity 接口
func (t LocalTag) GetUpdateTime() int64 {
	return t.UpdateTime
}

// SetCreateTime 实现 BaseEntity 接口
func (t LocalTag) SetCreateTime(time int64) {
	t.CreateTime = time
}

// SetUpdateTime 实现 BaseEntity 接口
func (t LocalTag) SetUpdateTime(time int64) {
	t.UpdateTime = time
}
